### PR TITLE
WT-7965 Update connection base write generation number at the end of recovery checkpoint

### PR DESCRIPTION
**Pull request by kommiharibabu**
During WiredTiger recovery, all the existing transaction ids are reset
based on the page write generation numbers present on the page after
rollback to stable is successfully finished. To avoid having transaction ids
on the pages that are written during recovery, WiredTiger uses the same
write generation number of the page that exists for the new page image
that is going write to disk. With this approach, when this page gets
loaded during the process of rollback to stable, it preserves the transaction
ids and resets only upon the next restart.

### CODE REVIEW

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -1534,7 +1535,11 @@ __rec_set_page_write_gen(WT_PAGE_HEADER *dsk, WT_BTREE *btree)
      * Other than salvage, the write generation number is used to reset the stale transaction id's
      * present on the page upon server restart.
      */
-    dsk->write_gen = ++btree->write_gen;
```
Comment (quchenhao) [inline src/reconcile/rec_write.c:1537]: Better to add a comment here to describe this.

Comment (kommiharibabu) [reply src/reconcile/rec_write.c]: I dropped the approach of using the same write gen number and update the connection base write generation number.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**